src/rulesets/Base/fastmath_able.jl

```diff
@@ -113,16 +113,16 @@ let
         function frule(
             (_, Δx, Δy),
             ::typeof(hypot),
-            x::T,
-            y::T,
-        ) where {T<:Union{Real,Complex}}
+            x::T1,
+            y::T2,
+        ) where {T1<:Union{Real,Complex}, T2<:Union{Real,Complex}}
```

Comment:
Is this signature just changed so that the tests in the loop pass? If so, maybe it's better to just take that test out of the loop. The signature we had here is the one that is FastMath compatible. There is a `hypot` that takes mixed arguments, but it promotes before calling this one, so currently we would just let the AD handle that. Same with the `rrule`.